src/Containers/SelectList.tsx

```diff
@@ -0,0 +1,183 @@
+import React, {
+    useState,
+    useCallback,
+    useMemo,
+    Children,
+    useLayoutEffect,
+    useRef,
+    isValidElement,
+    RefObject,
+} from 'react';
+import { scroll, position, transition, clickable, darken } from '@Utils/Mixins';
+
+import { LabelLayoutProps } from '@Layouts';
+import { useTransition } from '@Utils/Hooks';
+
+import styled from 'styled-components';
+
+const REVERSE = '-1';
+
+export interface SearchBarProps extends LabelLayoutProps {
+    value?: string | number;
+    onChange?: Function;
+    limit: number;
+    expanded: boolean;
+    inputValue?: string;
+}
+
+export const SelectList: React.FC<SearchBarProps> = ({
+    children,
+    limit,
+    expanded,
+    inputValue,
+    onChange = (): void => {},
+    name,
+}): React.ReactElement => {
+    const refSelectList = useRef() as RefObject<HTMLDivElement>;
+    const [numVisibleSelection, setNumVisibleSelection] = useState(limit);
+    const [animation] = useTransition(expanded);
+
+    useLayoutEffect((): void | (() => void) => {
+        if (refSelectList.current?.children.length) {
+            setNumVisibleSelection(refSelectList.current.children.length);
+        } else {
+            setNumVisibleSelection(0);
+        }
+    }, [expanded]);
+
+    const createList = (
+        thechildren: React.ReactNode[],
```

Comment:
isn't this supposed to be children?

Reply:
yes but since children is defined in props above we'll receive a typescript error so I changed it to thechildren in the function